src/main/java/org/spongepowered/api/util/blockray/BlockRay.java

```diff
@@ -790,8 +927,32 @@ public Vector3d position() {
     }
 
     /**
-     * Extends a filter by a number of blocks, regardless of what the extended
-     * filter does.
+     * A filter that will return {@code true} when the {@link BlockRayHit}
+     * represents <strong>any</strong> of the supplied {@link BlockType}s.
+     *
+     * @param types The type of block to allow
+     * @param <E> The extent to be applied in
+     * @return The filter instance
+     */
+    public static <E extends Extent> Predicate<BlockRayHit<E>> blockTypeFilter(final BlockType... types) {
+        return lastHit -> {
+            for (BlockType type : types) {
+                if (lastHit.getExtent().getBlockType(lastHit.getBlockX(), lastHit.getBlockY(), lastHit.getBlockZ()).equals(type)) {
```

Comment:
Move `lastHit.getExtent().getBlockType(lastHit.getBlockX(), lastHit.getBlockY(), lastHit.getBlockZ())` outside the loop

Reply:
Good spot.